apps/whois/mode/whois.pm

```diff
@@ -0,0 +1,308 @@
+        "custom-date-map:s"   => { name => 'custom_date_map', default => "DMY"}, # day month year
+    });
+    # This method will execute the command, but will not let me parse the outcome
+    return $self;
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::init(%options);
+    # Check domain has only two levels
+    my $domain_levels = split('\.', $self->{option_results}->{domain});
+    if ($domain_levels != 2){
+        $self->{output}->add_option_msg(
+            short_msg => "Wrong domain pattern '" . $self->{option_results}->{domain} . "'. Only DOMAIN.TLD supported");
+        $self->{output}->option_exit();
+    }
+    # Check custom date regex has at least 3 groups 
+    my $MinGroupCount = 3;
+    my $opening_groups = ($self->{option_results}->{custom_date_regex} =~ tr/\(//);
+    if ($opening_groups < $MinGroupCount) {
+        $self->{output}->add_option_msg(short_msg => "Error. At least 3 groups needed on regex. found: " . $opening_groups);
+        $self->{output}->option_exit();
+    }
+    my $closing_groups = ($self->{option_results}->{custom_date_regex} =~ tr/\)//);
+    if ($opening_groups != $closing_groups) {
+        $self->{output}->add_option_msg(short_msg => "Error. count of opening and closing parenthesis differ in regex: " . $opening_groups .'<>'. $closing_groups);
+        $self->{output}->option_exit();
+    }
+    # Check custom date map has same number of groups as the regex, and only contains the Characters YMDhms
+    unless ($self->{option_results}->{custom_date_map} =~ /Y/ && $self->{option_results}->{custom_date_map} =~ /D/ && $self->{option_results}->{custom_date_map} =~ /M/) {
+        $self->{output}->add_option_msg(short_msg => "Error. custom-date-map must contaion Y, M and D characters.");
+        $self->{output}->option_exit();
+    }
+    if (($self->{option_results}->{custom_date_map} =~ tr/YMDhms//) > $opening_groups) {
+        $self->{output}->add_option_msg(short_msg => "custom-date-map expects more groups than the count defined in custom-date-regex");
+        $self->{output}->option_exit();
+    }
+
+
+    if (($self->{perfdata}->threshold_validate(label => 'warning', value => $self->{option_results}->{warning})) == 0) {
+        $self->{output}->add_option_msg(short_msg => "Wrong warning threshold '" . $self->{option_results}->{warning} . "'.");
+        $self->{output}->option_exit();
+    }
+    if (($self->{perfdata}->threshold_validate(label => 'critical', value => $self->{option_results}->{critical})) == 0) {
+        $self->{output}->add_option_msg(short_msg => "Wrong critical threshold '" . $self->{option_results}->{critical} . "'.");
+        $self->{output}->option_exit();
+    }
+}
+
+sub run {
+    my ($self, %options) = @_;
+    my $catch_error_message = '';
+    # Extracts expiry date
+    my ($lerror, $stdout, $exit_code) = centreon::plugins::misc::backtick(
```

Comment:
Please try to use the execute_command from the custom_mode template like we do here: https://github.com/centreon/centreon-plugins/blob/2fd7bbb451f3dbf44b193d3566649bd0c2ed8c3e/os/linux/local/mode/uptime.pm#L60

Reply:
```
UNKNOWN: Can't call method "execute_command" on an undefined value at /home/pesantos/Repos/centreon-plugins/apps/whois/cli/mode/whois.pm line 116.
```
help sought...